src/main/java/de/hysky/skyblocker/skyblock/chocolatefactory/ChocolateFactorySolver.java

```diff
@@ -0,0 +1,165 @@
+package de.hysky.skyblocker.skyblock.chocolatefactory;
+
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.utils.ItemUtils;
+import de.hysky.skyblocker.utils.render.gui.ColorHighlight;
+import de.hysky.skyblocker.utils.render.gui.ContainerSolver;
+import it.unimi.dsi.fastutil.ints.*;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.text.Text;
+import org.apache.commons.math3.analysis.function.Min;
+
+import java.util.*;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class ChocolateFactorySolver extends ContainerSolver {
+	private static final Pattern CPS_PATTERN = Pattern.compile("([\\d,.]+) Chocolate per second");
+	private static final Pattern CPS_INCREASE_PATTERN = Pattern.compile("\\+([\\d,]+) Chocolate per second");
+	private static final Pattern COST_PATTERN = Pattern.compile("Cost ([\\d,]+) Chocolate");
+	private static final Pattern TOTAL_MULTIPLIER_PATTERN = Pattern.compile("Total Multiplier: ([\\d.]+)x");
+	private static final Pattern MULTIPLIER_INCREASE_PATTERN = Pattern.compile("\\+([\\d.]+)x Chocolate per second");
+	private static final Pattern TOTAL_CHOCOLATE_PATTERN = Pattern.compile("([\\d,]+) Chocolate");
+
+	public ChocolateFactorySolver() {
+		super("^Chocolate Factory$");
+	}
+
+	@Override
+	protected boolean isEnabled() {
+		return SkyblockerConfigManager.get().helpers.chocolateFactory.enableChocolateFactoryHelper;
+	}
+
+	@Override
+	protected List<ColorHighlight> getColors(String[] groups, Int2ObjectMap<ItemStack> slots) {
+		final Int2ObjectMap<Rabbit> cpsIncreaseFactors = new Int2ObjectLinkedOpenHashMap<>(6);
+		for (int i = 29; i <= 33; i++) { // The 5 rabbits slots are in 29, 30, 31, 32 and 33.
+			ItemStack item = slots.get(i);
+			if (item.getItem() != Items.PLAYER_HEAD || item.isEmpty()) continue;
+
+			int finalI = i; //Java, pfft.
+			getRabbit(item).ifPresent(rabbit -> cpsIncreaseFactors.put(finalI, rabbit));
```

Comment:
```suggestion
			if (item.isOf(Items.PLAYER_HEAD)) {
				int finalI = i; //Java, pfft.
				getRabbit(item).ifPresent(rabbit -> cpsIncreaseFactors.put(finalI, rabbit));
			}
```
remove the useless empty check, use isOf instead of == and invert the if logic